frontend/src/app/components/janusVideoRoom/janusVideoRoom.service.ts

```diff
@@ -422,37 +471,64 @@ export class JanusVideoRoomService {
     }
   }
 
-  // Forward stream to janus port
-  forwardRemoteFeed(login: string, port: number, forwardedCallback: (login: string) => void) {
-    if (!(login in this.remoteHandles)) {
-      this.toastr.error(`Could not find remote handle for ${login}`);
-      return;
-    }
-
-    // Stop (if exists) => Start => Update state => Callback.
-    this.$http.get(this.config.backendUri + '/shidur_state').success((shidurState: IShidurState) => {
-      if (!shidurState.janus) {
-        shidurState.janus = <any>{};
-      }
-      if (!shidurState.janus.portsFeedForwardInfo) {
-        shidurState.janus.portsFeedForwardInfo = <any>{};
+  // API for client (channel) forward stream to janus port (sdi).
+  forwardRemoteFeed(login: string, port: number): ng.IPromise<any> {
+    // Returns promise.
+    return this.getAndUpdateShidurState((shidurState: IShidurState) => {
+      var deffered = this.$q.defer();
+      if (!(login in this.remoteHandles)) {
+        this.toastr.error(`Could not find remote handle for ${login}`);
```

Comment:
Yes, the user is the Sidur operator => maybe different text but a message should be displayed.